lms/djangoapps/grades/tests/test_grades.py

```diff
@@ -68,7 +69,14 @@ def test_all_empty_grades(self):
         """
         No students have grade entries.
         """
-        all_course_grades, all_errors = self._course_grades_and_errors_for(self.course, self.students)
+        with patch.object(
+            BlockStructureFactory,
+            'create_from_cache',
+            wraps=BlockStructureFactory.create_from_cache
+        ) as mock_create_from_cache:
+            all_course_grades, all_errors = self._course_grades_and_errors_for(self.course, self.students)
+            self.assertEquals(mock_create_from_cache.call_count, 1)
```

Comment:
It doesn't matter for the purpose of this optimization change.  I have verified that this test fails on the master branch.